src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -528,8 +538,20 @@ namespace xclcpuemhal2 {
         }
         if (auto sec = xrt_core::xclbin::get_axlf_section(top, ASK_GROUP_TOPOLOGY)) {
           memTopologySize = sec->m_sectionSize;
-          memTopology = new char[memTopologySize];
-          memcpy(memTopology, xclbininmemory + sec->m_sectionOffset, memTopologySize);
+          memTopology = std::unique_ptr<char[]>(new char[memTopologySize]);
```

Comment:
Simplify everywhere with
```suggestion
          memTopology = std::make_unique<char[]>(memTopologySize);
```

Reply:
@keryell , we cant do this. seeing below compilation error
.../src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:552:64: error: no matching function for call to ‘std::unique_ptr<char []>::unique_ptr(size_t&)’
           memTopology = std::unique_ptr<char[]>(memTopologySize);

Reply:
Which C++ language are you using?
It looks like it is in C++14 https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique which is now 7 year old.

Reply:
No, @keryell .  std::make_unique does not support char[] for at least some of the compilers in the distributions we support.  I don't like std::unique_ptr<char[]>  in general, I prefer std::string or std::vector, but that is besides the point.